src/components/page-components/ExplorePage/ExplorePage.tsx

```diff
@@ -9,25 +9,44 @@ import { sendToBackground } from "@plasmohq/messaging"
 import BurgerMenu from "~components/BurgerMenu/BurgerMenu"
 import FilterTags from "~components/FilterTags/FilterTags"
 import Footer from "~components/Footer/Footer"
+import PaginationNav from "~components/PaginationNav/PaginationNav"
 import ProjectCard from "~components/ProjectCards/ProjectCard"
-import type { AllProjectResponse } from "~types/projectTypes"
+import type { Meta } from "~types/baseTypes"
+import type { ProjectData } from "~types/projectTypes"
 
 export default function ExplorePage() {
-  const [projects, setProjects] =
-    useState<AllProjectResponse["data"]>()
+  const [projects, setProjects] = useState<ProjectData[]>()
+  const [pageNumber, setPageNumber] = useState<number>(1)
+  const [pageCount, setPageCount] = useState<number>(1)
   const { showBoundary } = useErrorBoundary()
 
   useEffect(() => {
     const fetchAllProjects = async () => {
-      const { data, error } = await sendToBackground({
-        name: "fetchAllProjects"
-      })
+      const {
+        data,
+        error,
+        meta
+      }: { data: ProjectData[]; error: string | null; meta: Meta } =
+        await sendToBackground({
+          name: "fetchAllProjects",
+          body: { page: pageNumber }
+        })
 
       if (error) showBoundary(error)
+        
+      setPageCount(meta.pagination.pageCount)
       setProjects(data)
     }
     fetchAllProjects()
-  }, [])
+  }, [pageNumber])
+
+  const navigateToNext = () => {
+    setPageNumber(page => page + 1)
```

Comment:
Another small thing. If you move the click handlers to the PaginationNav component and just pass the setPageNumber function you would not have to declare the functions in both the explore and favourites page prevention a little bit of code duplication.

Reply:
uhh yeah, that looks way better